network-config-analyzer/PeerContainer.py

```diff
@@ -172,6 +146,47 @@ def _set_peer_list(self, peer_resources_list, config_name):
         print(f'{config_name}: cluster has {self.get_num_peers()} unique endpoints, '
               f'{self.get_num_namespaces()} namespaces')
 
+    def set_services_and_populate_target_pods(self, srv_list):
+        """
+        Populates services from the given service list,
+        and for every service computes and populates its target pods.
+        :param list srv_list: list of service in K8sService format
+        :return: None
+        """
+        for srv in srv_list:
+            for key, val in srv.selector.items():
+                srv.target_pods |= self.get_peers_with_label(key, [val], GenericYamlParser.FilterActionType.In,
+                                                             srv.namespace)
+            # remove target_pods that don't contain named ports referenced by target_ports
+            for port in srv.ports.values():
+                if not isinstance(port.target_port, str):
+                    continue
+                # check if all pods include this named port, and remove those that don't
+                pods_to_remove = PeerSet()
+                for pod in srv.target_pods:
+                    pod_named_port = pod.named_ports.get(port.target_port)
+                    if not pod_named_port:
+                        print(f'Warning: The named port {port.target_port} referenced in Service {srv.name}'
+                              f' is not defined in the pod {pod}. Ignoring the pod')
+                        pods_to_remove.add(pod)
+                srv.target_pods -= pods_to_remove
+
+            self.services[srv.name] = srv
+
+    def get_service(self, srv_name):
```

Comment:
it seems that `get_service `and `get_services `are not in use

Reply:
removed